RELEASE.md

```diff
@@ -0,0 +1,53 @@
+# Boxo Releases
+
+## Release Policies
+### Versioning
+Boxo only releases minor and patch versions, and has no plans to change major versions from v0.
+
+Minor versions contain new or removed functionality, or significant changes in dependencies.
+
+Patch versions contain "fixes", which are generally small, targeted, and involve limited changes in functionality (such as bug fixes, security fixes, etc.). The amount of backporting depends on the severity of the issue and the impact on users...most non-critical fixes won't be backported.
+
+As a result, Boxo maintainers recommend that consumers stay up-to-date with Boxo releases.
+
+### Go Compatibility
+At any given point, the Go team supports only the latest two versions of Go released (see https://go.dev/doc/devel/release). Boxo maintainers will strive to maintain compatibilty with the older of the two supported versions, so that Boxo is also compatible with the latest two versions of Go.
+
+### Release Criteria
+Boxo does not follow a regular release cadence. Releases are performed on-demand as significant changes are made (new features added, refactorings, deprecations, etc.). The [release process](#release-process) below describes how maintainers perform a release.
+
+### Testing
+Boxo maintainers are in the process of moving tests into Boxo from Kubo. Until this is completed, Boxo releases rely on Kubo tests for extra confidence. So releases must include plumbing Boxo into Kubo and ensuring its tests pass. See [Release Process](#release-process) for details.
+
+### Changelogs
+Boxo follows the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format. New commits should add changelog entries into the `[Unreleased]` section so that there is no changelog scramble when a new version needs to be released.
+
+We intend to use the standardized changelog to automate releases by onboarding with [Changelog Driven Release](https://github.com/pl-strflt/changelog-driven-release).
+
+## Release Process
+Create an issue for the version, if it doesn't already exist, then copy-paste the following checklist into the top of the issue:
+
+- [ ] Verify your [GPG signature](https://docs.github.com/en/authentication/managing-commit-signature-verification) is configured in local git and GitHub
+- [ ] Ensure Boxo and Kubo are checked out on your system
+- [ ] Create a release branch called `release-vX.Y.Z` and push it
+  * Use `master` as base if `Z == 0`
+  * Use `release` as base if `Z > 0`
+- [ ] Tidy the changelog and push any changelog changes to `release-vX.Y.Z`
+- [ ] Create a draft PR merging `release-vX.Y.Z` into `release`
```

Comment:
```suggestion
- [ ] Create a draft PR merging `release-vX.Y.Z` into `release`
   - [ ] Reference this release issue
```

Maybe also give guidance on how to name the PR (relates to https://github.com/ipfs/boxo/pull/264#issuecomment-1498496613 )